### PR TITLE
Update legacy oracle proposals to new proposal formats using msg server

### PR DESCRIPTION
**Pull request by jelysn**
- Update legacy oracle proposals to new proposal formats using msg server
- Add migration script to migrate eden vesting to 12hrs
- Update joinPool & exitPool sender to be accepted by wasmbinding broker address

### CODE REVIEW

File: x/amm/client/wasm/msg_exit_pool.go [hunk 1]
```diff
@@ -15,7 +15,8 @@ func (m *Messenger) msgExitPool(ctx sdk.Context, contractAddr sdk.AccAddress, ms
 		return nil, nil, wasmvmtypes.InvalidRequest{Err: "exit pool null msg"}
 	}
 
-	if msg.Sender != contractAddr.String() {
+	brokerAddress := m.parameterKeeper.GetParams(ctx).BrokerAddress
+	if msg.Sender != contractAddr.String() && contractAddr.String() != brokerAddress {
```
[the remark this answers is not in the document]
Comment (cosmic-vagabond) [reply x/amm/client/wasm/msg_exit_pool.go:19]: actually it makes sense to keep the condition as it is as we are not using any contract tokens for this message

File: x/amm/client/wasm/msg_join_pool.go [hunk 2]
```diff
@@ -15,7 +15,8 @@ func (m *Messenger) msgJoinPool(ctx sdk.Context, contractAddr sdk.AccAddress, ms
 		return nil, nil, wasmvmtypes.InvalidRequest{Err: "join pool null msg"}
 	}
 
-	if msg.Sender != contractAddr.String() {
+	brokerAddress := m.parameterKeeper.GetParams(ctx).BrokerAddress
+	if msg.Sender != contractAddr.String() && contractAddr.String() != brokerAddress {
```
[the remark this answers is not in the document]
Comment (cosmic-vagabond) [reply x/amm/client/wasm/msg_join_pool.go:19]: same here it makes sense to keep the condition expression as it is

File: x/oracle/types/message_proposals.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgRemovePriceFeeders) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Authority)
+	if err != nil {
+		return errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid feeder address (%s)", err)
+	}
+	return nil
+}
```
Comment (cosmic-vagabond) [inline x/oracle/types/message_proposals.go:181]: @jelysn those changes causes the gov proposals prior to 114 to fail to display as some proposals are using old proposal structure.

File: x/amm/client/wasm/msg_join_pool.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jelysn) [reply x/amm/client/wasm/msg_join_pool.go:19]:
```
	brokerAddress := m.parameterKeeper.GetParams(ctx).BrokerAddress
	if contractAddr.String() != brokerAddress {
		return nil, nil, wasmvmtypes.InvalidRequest{Err: "contract address must be broker address"}
	}

	if msgOpen.Creator != contractAddr.String() {
		return nil, nil, wasmvmtypes.InvalidRequest{Err: "open wrong sender"}
	}
```
This is a different meaning from original purpose. `msg.Sender != contractAddr.String()` is the case when contract open position with its own token, and `contractAddr.String() != brokerAddress` is the case when the contract just open the position for a user.

File: x/oracle/types/message_proposals.go (hunk 3 above)
Comment (cosmic-vagabond) [reply x/oracle/types/message_proposals.go:213]: fixing this by removing old proposals

**Review by cosmic-vagabond — APPROVED** [review]

**Comment by cosmic-vagabond** [comment]
software upgrade test ✅ 

```
2:54PM ERR CONSENSUS FAILURE!!! err="UPGRADE \"v0.29.21\" NEEDED at height: 6278056: " module=consensus stack="goroutine 42 [running]:\nruntime/debug.Stack()\n\truntime/debug/stack.go:24 +0x64\ngithub.com/cometbft/cometbft/consensus.(*State).receiveRoutine.func2()\n\tgithub.com/cometbft/cometbft@v0.37.2/consensus/state.go:732 +0x44\npanic({0x1077e40c0, 0x14000d6fde0})\n\truntime/panic.go:884 +0x204\ngithub.com/cosmos/cosmos-sdk/x/upgrade.BeginBlocker(_, {{0x107ccbe40, 0x140002f0080}, {0x107ce2130, 0x1400a9eeb80}, {{0xb, 0x0}, {0x1400631ab00, 0xd}, 0x5fcba8, ...}, ...}, ...)\n\tgithub.com/cosmos/cosmos-sdk@v0.47.4/x/upgrade/abci.go:78 +0xc28\ngithub.com/cosmos/cosmos-sdk/x/upgrade.AppModule.BeginBlock(...)\n\tgithub.com/cosmos/cosmos-sdk@v0.47.4/x/upgrade/module.go:161\ngithub.com/cosmos/cosmos-sdk/types/module.(*Manager).BeginBlock(_, {{0x107ccbe40, 0x140002f0080}, {0x107ce2130, 0x1400a9eeb80}, {{0xb, 0x0}, {0x1400631ab00, 0xd}, 0x5fcba8, ...}, ...}, ...)\n\tgithub.com/cosmos/cosmos-sdk@v0.47.4/types/module/module.go:562 +0x170\ngithub.com/elys-network/elys/app.(*ElysApp).BeginBlocker(...)\n\tgithub.com/elys-network/elys/app/app.go:1343\ngithub.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).BeginBlock(_, {{0x1400cfc20a0, 0x20, 0x20}, {{0xb, 0x0}, {0x1400631ab00, 0xd}, 0x5fcba8, {0x2eca8018, ...}, ...}, ...})\n\tgithub.com/cosmos/cosmos-sdk@v0.47.4/baseapp/abci.go:196 +0x674\ngithub.com/cometbft/cometbft/abci/client.(*localClient).BeginBlockSync(_, {{0x1400cfc20a0, 0x20, 0x20}, {{0xb, 0x0}, {0x1400631ab00, 0xd}, 0x5fcba8, {0x2eca8018, ...}, ...}, ...})\n\tgithub.com/cometbft/cometbft@v0.37.2/abci/client/local_client.go:283 +0x108\ngithub.com/cometbft/cometbft/proxy.(*appConnConsensus).BeginBlockSync(_, {{0x1400cfc20a0, 0x20, 0x20}, {{0xb, 0x0}, {0x1400631ab00, 0xd}, 0x5fcba8, {0x2eca8018, ...}, ...}, ...})\n\tgithub.com/cometbft/cometbft@v0.37.2/proxy/app_conn.go:101 +0xf4\ngithub.com/cometbft/cometbft/state.execBlockOnProxyApp({0x107ccd3b0?, 0x14006f20de0}, {0x107cde7c0, 0x14000a288e8}, 0x1400effe3c0, {0x107ce2d38, 0x14001296108}, 0x5fcba7?)\n\tgithub.com/cometbft/cometbft@v0.37.2/state/execution.go:363 +0x3e8\ngithub.com/cometbft/cometbft/state.(*BlockExecutor).ApplyBlock(_, {{{0xb, 0x0}, {0x14006ee17da, 0x6}}, {0x14006ee17f0, 0xd}, 0x5fcba0, 0x5fcba7, {{0x140094f6800, ...}, ...}, ...}, ...)\n\tgithub.com/cometbft/cometbft@v0.37.2/state/execution.go:197 +0xe4\ngithub.com/cometbft/cometbft/consensus.(*State).finalizeCommit(0x140016c8e00, 0x5fcba8)\n\tgithub.com/cometbft/cometbft@v0.37.2/consensus/state.go:1700 +0x964\ngithub.com/cometbft/cometbft/consensus.(*State).tryFinalizeCommit(0x140016c8e00, 0x5fcba8)\n\tgithub.com/cometbft/cometbft@v0.37.2/consensus/state.go:1609 +0x28c\ngithub.com/cometbft/cometbft/consensus.(*State).enterCommit.func1()\n\tgithub.com/cometbft/cometbft@v0.37.2/consensus/state.go:1544 +0xa0\ngithub.com/cometbft/cometbft/consensus.(*State).enterCommit(0x140016c8e00, 0x5fcba8, 0x0)\n\tgithub.com/cometbft/cometbft@v0.37.2/consensus/state.go:1582 +0xb90\ngithub.com/cometbft/cometbft/consensus.(*State).addVote(0x140016c8e00, 0x14003d8a6e0, {0x0, 0x0})\n\tgithub.com/cometbft/cometbft@v0.37.2/consensus/state.go:2212 +0x1790\ngithub.com/cometbft/cometbft/consensus.(*State).tryAddVote(0x140016c8e00, 0x14003d8a6e0, {0x0?, 0x1042c6d40?})\n\tgithub.com/cometbft/cometbft@v0.37.2/consensus/state.go:2001 +0x28\ngithub.com/cometbft/cometbft/consensus.(*State).handleMsg(0x140016c8e00, {{0x107c9efe0, 0x1400093e1c0}, {0x0, 0x0}})\n\tgithub.com/cometbft/cometbft@v0.37.2/consensus/state.go:861 +0x37c\ngithub.com/cometbft/cometbft/consensus.(*State).receiveRoutine(0x140016c8e00, 0x0)\n\tgithub.com/cometbft/cometbft@v0.37.2/consensus/state.go:788 +0x350\ncreated by github.com/cometbft/cometbft/consensus.(*State).OnStart\n\tgithub.com/cometbft/cometbft@v0.37.2/consensus/state.go:379 +0xf4\n"
2:54PM INF service stop impl={"Logger":{"Logger":{}}} module=consensus msg={} wal=/tmp/elys/data/cs.wal/wal
2:54PM INF service stop impl={"Dir":"/tmp/elys/data/cs.wal","Head":{"ID":"pC3p6Sa4uPgZ:/tmp/elys/data/cs.wal/wal","Path":"/tmp/elys/data/cs.wal/wal"},"ID":"group:pC3p6Sa4uPgZ:/tmp/elys/data/cs.wal/wal","Logger":{"Logger":{}}} module=consensus msg={} wal=/tmp/elys/data/cs.wal/wal
2024/03/06 14:54:39 Block height 6278056 reached
2:54PM INF Ensure peers module=pex numDialing=0 numInPeers=0 numOutPeers=0 numToDial=10
2:54PM INF No addresses to dial. Falling back to seeds module=pex
2:54PM INF Timed out dur=3000 height=6278056 module=consensus round=0 step=3
2024/03/06 14:54:44 Process killed successfully
2024/03/06 14:54:44 Process was killed: signal: killed
2024/03/06 14:54:49 Waiting for node to start...
2:54PM INF starting node with ABCI Tendermint in-process module=server
2:54PM INF service start impl=multiAppConn module=proxy msg={}
2:54PM INF service start connection=query impl=localClient module=abci-client msg={}
2:54PM INF service start connection=snapshot impl=localClient module=abci-client msg={}
2:54PM INF service start connection=mempool impl=localClient module=abci-client msg={}
2:54PM INF service start connection=consensus impl=localClient module=abci-client msg={}
2:54PM INF service start impl=EventBus module=events msg={}
2:54PM INF service start impl=PubSub module=pubsub msg={}
2:54PM INF service start impl=IndexerService module=txindex msg={}
2:54PM INF ABCI Handshake App Info hash="�Ի\x7f��4\"�\x18g����6|\naUk�1����B�<�~" height=6278055 module=consensus protocol-version=0 software-version=v0.29.21
2:54PM INF ABCI Replay Blocks appHeight=6278055 module=consensus stateHeight=6278055 storeHeight=6278056
2:54PM INF Replay last block using real app module=consensus
2:54PM INF applying upgrade "v0.29.21" at height: 6278056 module=server
2:54PM INF Running upgrade handler for v0.29.21 module=server
2:54PM INF Deleting proposals with ID <= 113 module=server
2:54PM INF migrating module commitment from version 1 to version 2 module=server
2:54PM INF migrating module oracle from version 2 to version 3 module=server
```

**Pull request merged**